File: gluecodium/src/test/resources/smoke/basic_types/output/android/com/example/smoke/BasicTypes.java
/*
 *
 */
package com.example.smoke;
import android.support.annotation.NonNull;
import com.example.NativeBase;
public final class BasicTypes extends NativeBase {
    /**
     * For internal use only.
     * @exclude
     */
    protected BasicTypes(final long nativeHandle, final Object dummy) {
        super(nativeHandle, new Disposer() {
            @Override
            public void disposeNative(long handle) {
                disposeNativeHandle(handle);
            }
        });
    }
    private static native void disposeNativeHandle(long nativeHandle);
    @NonNull
    public static native String stringFunction(@NonNull final String input);
    public static native boolean boolFunction(final boolean input);
    public static native float floatFunction(final float input);
    public static native double doubleFunction(final double input);
    public static native byte byteFunction(final byte input);
    public static native short shortFunction(final short input);
    public static native int intFunction(final int input);
    public static native long longFunction(final long input);
    public static native short ubyteFunction(final short input);
    public static native int ushortFunction(final int input);
    public static native long uintFunction(final long input);
    public static native long ulongFunction(final long input);
}

File: gluecodium/src/test/resources/smoke/basic_types/output/android-kotlin/com/example/smoke/BasicTypes.kt
/*

 *
 */

package com.example.smoke

import com.example.NativeBase

class BasicTypes : NativeBase {



    /*
     * For internal use only.
     * @hidden
     * @param nativeHandle The handle to resources on C++ side.
     * @param tag Tag used by callers to avoid overload resolution problems.
     */
    protected constructor(nativeHandle: Long, tag: Any?)
        : super(nativeHandle, { disposeNativeHandle(it) }) {}




    companion object {
        @JvmStatic private external fun disposeNativeHandle(nativeHandle: Long)
        @JvmStatic external fun stringFunction(input: String) : String
        @JvmStatic external fun boolFunction(input: Boolean) : Boolean
        @JvmStatic external fun floatFunction(input: Float) : Float
        @JvmStatic external fun doubleFunction(input: Double) : Double
        @JvmStatic external fun byteFunction(input: Byte) : Byte
        @JvmStatic external fun shortFunction(input: Short) : Short
        @JvmStatic external fun intFunction(input: Int) : Int
        @JvmStatic external fun longFunction(input: Long) : Long
        @JvmStatic external fun ubyteFunction(input: Short) : Short
        @JvmStatic external fun ushortFunction(input: Int) : Int
        @JvmStatic external fun uintFunction(input: Long) : Long
        @JvmStatic external fun ulongFunction(input: Long) : Long
    }
}
